### PR TITLE
User settings

### PR DESCRIPTION
**Pull request by CupOfGeo**

### CODE REVIEW

File: src/text/text_editor.py [hunk 1]
```diff
@@ -46,16 +50,50 @@ class ApplicationState:
...
+                j.write(default_user_settings)
+
+        self.show_status_bar = True
+        if "last_path" in self.user_settings and self.user_settings["last_path"]:
+            self.current_path = self.user_settings["last_path"]
+        else:
+            self.current_path = None
+
+
+appState = ApplicationState()
```
Comment (BoraxTheClean) [inline src/text/text_editor.py]: We need to move this to a different file. We can't have global variables like this all over the place.

File: src/text/text_editor.py [hunk 2]
```diff
@@ -46,16 +50,50 @@ class ApplicationState:
...
+
+
+# class UserSettings():
+#     """User Settings object"""
+#     def __init__(self):
+#         self.last_path = ""
+#         self.style = ""
```
Comment (BoraxTheClean) [inline src/text/text_editor.py]:
```suggestion
```

Delete

File: src/text/text_editor.py [hunk 3]
```diff
@@ -194,54 +234,118 @@ def __pt_container__(self):
...
+    """Paste with Ctrl-V"""
+    do_paste()
+
+
+@bindings.add("c-z")
+def undo_changes(event: KeyPressEvent) -> None:
+    """Undo with Ctrl-Z"""
+    do_undo()
+
+
```
Comment (BoraxTheClean) [inline src/text/text_editor.py]: I think you need to merge main into this file and fix conflicts, these are already on main. Your branch probably wasn't up to date when you started this feature.

File: src/text/text_editor.py [hunk 4]
```diff
@@ -194,54 +234,118 @@ def __pt_container__(self):
...
 
 def do_scroll_menu() -> None:
     """Open Scroll Menu"""
-    show_scroll("Scroll", "buf")
+    show_scroll("Notes", "buf")
 
 
 def show_scroll(title: str, text: str) -> None:
-    """Shows about message"""
+    """Shows a MessageDialog with a certain title and text"""
```
Comment (BoraxTheClean) [inline src/text/text_editor.py]: 💯

File: src/text/text_editor.py [hunk 5]
```diff
@@ -252,7 +356,7 @@ async def coroutine() -> None:
 
 def do_about() -> None:
     """About from menu select"""
-    show_message("About", "Text editor demo.\nCreated by Jonathan Slenders.")
+    show_message("About", "ThoughtBox\nCreated by Adaptable Antelopes.")
```
Comment (BoraxTheClean) [inline src/text/text_editor.py]: Represent ✊

File: src/text/text_editor.py [hunk 6]
```diff
@@ -285,14 +389,20 @@ async def show_dialog_as_float(dialog: PopUpDialog) -> None:
...
-    """Doesn't make new file just clears text_field but i guess thats fine till save"""
+    """Makes a new file"""
     text_field.text = ""
+    set_current_path(None)
+    set_title("ThoughtBox - Untitled")
 
 
 def do_exit() -> None:
     """Exit"""
+    with open("user_setting.json", "w") as j:
```
Comment (BoraxTheClean) [inline src/text/text_editor.py]:
```suggestion
    with open(".user_setting.json", "w") as j:
```
Let's make this a hidden file.

File: src/text/text_editor.py (hunk 1 above)
Comment (SharpBit) [reply src/text/text_editor.py]: Also, should be in snake_case, not camelCase

File: src/text/text_editor.py (hunk 4 above)
Comment (SharpBit) [reply src/text/text_editor.py]: this is already on main too lol

File: src/application/state.py [hunk 7]
```diff
@@ -6,6 +9,24 @@ class ApplicationState:
...
 
+    # def __init__(self):
+    #     self.show_status_bar = True
+    #     self.current_path = None
+
     def __init__(self):
-        self.show_status_bar = True
         self.current_path = None
+        try:
+            with open("user_setting.json", "r") as j:
```
Comment (BoraxTheClean) [inline src/application/state.py]: Please make this a hidden file in the `.thought_box` directory.

**Review by BoraxTheClean — CHANGES_REQUESTED** [review]

**Review by BoraxTheClean — COMMENTED** [review]
Meant to just comment. Someone else can approve, I won't have time to review this for a few hours.

File: src/text/text_editor.py (hunk 5 above)
Comment (CupOfGeo) [reply src/text/text_editor.py]: another one thats not mine but i pulled it from somewhere

File: src/application/state.py [hunk 8]
```diff
@@ -6,6 +9,24 @@ class ApplicationState:
...
+                self.user_settings = json.loads(j.read())
+        except FileNotFoundError:
+            # if for some reason the file is deleted but then i need to maintain all the setting here
+            self.user_settings = {"last_path": None, "style": ""}  # color picker?
```
Comment (ghost) [inline src/application/state.py:27]: Does an empty string as a style value just get ignored?

File: src/navigation/menu_bar.py [hunk 9]
```diff
@@ -89,14 +109,32 @@ def do_save_as_file(self) -> None:
...
+            # 2. Doesn't consist exclusively of whitespace
+            # 3. Isn't the string "." or ".."
+            # 4. if the file already exists warning of over writing
+            if (
+                user_entered_path
+                and not str.isspace(user_entered_path)
+                and user_entered_path not in [".", ".."]
+                and os.path.isfile(user_entered_path)
+            ):
+                path = NOTES_DIR + "/" + user_entered_path
```
Comment (ghost) [inline src/navigation/menu_bar.py]: Would it be better to use `os.path.join` here? I think forward slashes are okay here, even if the user's machine is Windows.

File: src/navigation/menu_bar.py [hunk 10]
```diff
@@ -116,6 +154,12 @@ def do_new_file(self) -> None:
 
     def do_exit(self) -> None:
         """Exit"""
+        with open("user_setting.json", "w") as j:
```
Comment (ghost) [inline src/navigation/menu_bar.py]: This is a very handy feature.

File: README.md [hunk 11]
```diff
@@ -7,22 +7,24 @@ Free your self from distractions with our minimalists thought journal and note-t
 
 ## How it works
 
-Running this program <how do they run the program python text_editor.py?> Will turn your terminal
+Running this program <how do they run the program $python src/application_entry.py?> Will turn your terminal
 into your very own ThoughtBox where you are free to take your notes however you like as long as its ascii approved.
```
Comment (ghost) [inline README.md:11]: This could use proofreading, but that's a minor detail.

**Review by ghost — COMMENTED** [review]
Good work. I think the reorganized save function is easier to understand, and the user settings json is a handy feature.

**Review by BoraxTheClean — APPROVED** [review]

**Pull request merged**